projects/kit/components/input-number/input-number.component.ts

```diff
@@ -0,0 +1,255 @@
+import {MaskitoDirective} from '@maskito/angular';
+import type {MaskitoOptions} from '@maskito/core';
+import {maskitoInitialCalibrationPlugin} from '@maskito/core';
+import {
+    maskitoCaretGuard,
+    maskitoNumberOptionsGenerator,
+    maskitoParseNumber,
+} from '@maskito/kit';
+import {TuiControl} from '@taiga-ui/cdk/classes';
+import {CHAR_HYPHEN, CHAR_MINUS} from '@taiga-ui/cdk/constants';
+import {TUI_IS_IOS, tuiFallbackValueProvider} from '@taiga-ui/cdk/tokens';
+import {tuiInjectElement} from '@taiga-ui/cdk/utils/dom';
+import {tuiIsSafeToRound} from '@taiga-ui/cdk/utils/math';
+import {TuiTextfieldDirective} from '@taiga-ui/core/components/textfield';
+import {TUI_DEFAULT_NUMBER_FORMAT, TUI_NUMBER_FORMAT} from '@taiga-ui/core/tokens';
+import {tuiFormatNumber} from '@taiga-ui/core/utils/format';
+import {tuiMaskito} from '@taiga-ui/kit/utils';
+
+import {TUI_INPUT_NUMBER_OPTIONS} from './input-number.options';
+
+const DEFAULT_MAX_LENGTH = 18;
+
+@Component({
+    standalone: true,
+    selector: 'input[tuiInputNumber]',
+    template: '',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [tuiFallbackValueProvider(null)],
+    hostDirectives: [
+        // TODO: replace with `TuiWithTextfield` after merging of https://github.com/taiga-family/taiga-ui/pull/9976
+        {
+            directive: TuiTextfieldDirective,
+            inputs: ['invalid', 'focused', 'readOnly', 'state'],
+        },
+        MaskitoDirective,
+    ],
+    host: {
+        '[value]': 'textfieldValue()',
+        '[disabled]': 'disabled()',
+        '[attr.inputMode]': 'inputMode()',
+        '[attr.maxLength]': 'maxLength()',
+        '(input)': 'onInput()',
+        '(blur)': 'onBlur()',
+        '(focus)': 'onFocus()',
+    },
+})
+export class TuiInputNumber extends TuiControl<number | null> {
+    private readonly element = tuiInjectElement<HTMLInputElement>();
+    private readonly isIOS = inject(TUI_IS_IOS);
+    private readonly options = inject(TUI_INPUT_NUMBER_OPTIONS);
+    private readonly numberFormat = toSignal(inject(TUI_NUMBER_FORMAT), {
+        initialValue: TUI_DEFAULT_NUMBER_FORMAT,
+    });
+
+    private readonly min = signal(this.options.min);
+    private readonly max = signal(this.options.max);
+    private readonly prefix = signal(this.options.prefix);
+    private readonly postfix = signal(this.options.postfix);
+
+    private readonly precision = computed(() =>
+        Number.isNaN(this.numberFormat().precision) ? 2 : this.numberFormat().precision,
+    );
+
+    private readonly isIntermediateState = computed(() => {
+        const value = maskitoParseNumber(
+            this.textfieldValue(),
+            this.numberFormat().decimalSeparator,
+        );
+
+        return value < 0 ? value > this.max() : value < this.min();
+    });
+
+    protected textfieldValue = signal(this.element.value || '');
+    protected override readonly transformer = this.options.valueTransformer;
```

Comment:
Maybe instead we should use DI:
```ts
{
  provide: TuiValueTransformer,
  useFactory: () => inject(TUI_INPUT_NUMBER_OPTIONS).valueTransformer`
}
```
If we plan to do this often with other components we can make a helper `tuiValueTransformerFrom(token)`

Reply:
https://github.com/taiga-family/taiga-ui/pull/10031/commits/f6354bee6b1718e61eca569a25e86ccfd835a29d